app/api/data.py

```diff
@@ -371,20 +376,29 @@ def edit_core_data_from_states_daily():
 
             flask.current_app.logger.info('Row for date not found, making new edit row: %s' % date)
             is_edited = True
+            new_rows.append(edited_core_data)
 
         else:
             # this row already exists, but check each value to see if anything changed. Easiest way
-            # to do this is to make a new CoreData and compare it to the existing one
+            # to do this is to make a new CoreData (edited_core_data) and compare it to the existing one
+            row_diffs = []
             for field in CoreData.__table__.columns.keys():
                 # we expect batch IDs to be different, skip comparing those
                 if field == 'batchId':
                     continue
                 # for any other field, compare away
-                if getattr(data_for_date, field) != getattr(edited_core_data, field):
+                old = getattr(data_for_date, field)
+                new = getattr(edited_core_data, field)
+                if old != new:
                     changed_fields_for_date.add(field)
                     is_edited = True
+                    row_diffs.append(ChangedValue(field=field, old=old, new=new))
+
+            # if the row has been edited, save the list of the differences
+            if is_edited:
```

Comment:
it might be more correct to check for `if row_diffs` instead, since you're only adding to `changed_rows` under that condition. Also, small nit - is the indentation off on 399?

Reply:
Good idea. Fixed and fixed.